src/app/static/src/app/components/Warning.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+    <div v-if="warnings.length > 0">
+        <h4 class="alert-heading pt-2">
+            <alert-triangle-icon size="1.5x" class="custom-class mr-1 mb-1"></alert-triangle-icon>
+            <span v-translate="headerText(origin)"></span>
+        </h4>
+        <!-- The outer styled divs are the visible window of warnings and inside them are the 
+        full window of warnings (the warningBox), which is always rendered as html but with 
+        its overflow hidden by the outer box. The size of the outer box dynamically adjusts 
+        depending on the size of the warningBox and whether the user has clicked show more or less. -->
+        <div :style="{ overflowY: 'hidden', height: `${this.renderedBoxHeight}px` }">
+            <ul class="mb-0" ref="warningBox" id="warningBox">
+                <li v-for="warning in warnings" :key="warning.text"><div :style="lineStyling">{{ warning.text }}</div></li>
+            </ul>
+            <!-- The below element will never be shown to the user but is used as a proxy to 
+            dynamically determine the height in pixels the warnings window should have.  -->
+            <p ref="line" class="mb-0 invisible">Hidden line</p>
+        </div>
+        <button @click="toggleShowFullBox" v-if="warningsLengthy" class="btn btn-link alert-link">{{ buttonText }}</button>
+    </div>                 
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import { AlertTriangleIcon } from "vue-feather-icons";
+    import i18next from "i18next";
+    import { mapStateProp } from "../utils";
+    import { RootState } from "../root";
+    import { Language } from "../store/translations/locales";
+    import { Warning } from "../generated";
+    import { Dict } from "../types";
+
+    interface Props {
+        origin: string;
+        warnings: Warning[];
+        maxLines: number;
+    }
+
+    interface Data {
+        showFullBox: boolean;
+        fullBoxHeight: number;
+        lineHeight: number;
+    }
+
+    interface Methods {
+        toggleShowFullBox: () => void;
+        updateDimensions: () => void;
+        headerText: (key: string) => string;
+    }
+
+    interface Computed  {
+        currentLanguage: Language;
+        renderedBoxHeight: number;
+        warningsLengthy: boolean;
+        buttonText: string;
+        maxBoxHeight: number;
+        lineStyling: {
+            height?: string
+        };
```

Comment:
This works for ts oddly enough but `Dict<string>` does not.

Reply:
`Dict<string | number | undefined>` worked for me - number because of `maxLines` and I think it needs `undefined` because the keys are different depending on a conditional, though that seems a bit odd to me.